services/datamanager/builtin/capture/capture.go

```diff
@@ -164,9 +238,30 @@ func (c *Capture) Reconfigure(
 }
 
 // Close closes the capture manager.
-func (c *Capture) Close() {
+func (c *Capture) Close(ctx context.Context) {
 	c.FlushCollectors()
 	c.closeCollectors()
+	c.mongoMU.Lock()
+	defer c.mongoMU.Unlock()
+	if c.mongoClient != nil {
+		c.logger.Info("closing mongo connection")
+		goutils.UncheckedError(c.mongoClient.Disconnect(ctx))
+		c.mongoClient = nil
+		c.mongoCollection = nil
+		c.mongoConfig = nil
+	}
+}
+
+func (c *Capture) closeNoMongoMutex(ctx context.Context) {
```

Comment:
is this the same as `Close()` L241?

Reply:
no, close takes a mutex. this does not, hence the `No...Mutex` in the name.

Reply:
oops :) thx